kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ClusterManagerImpl.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableList;
+
+import io.confluent.kafkarest.entities.Broker;
+import io.confluent.kafkarest.entities.Cluster;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.DescribeClusterOptions;
+import org.apache.kafka.clients.admin.DescribeClusterResult;
+import org.apache.kafka.common.KafkaFuture;
+
+final class ClusterManagerImpl implements ClusterManager {
+
+  private final Admin adminClient;
+
+  @Inject
+  ClusterManagerImpl(Admin adminClient) {
+    this.adminClient = Objects.requireNonNull(adminClient);
+  }
+
+  @Override
+  public CompletableFuture<List<Cluster>> listClusters() {
+    DescribeClusterResult describeCluster =
```

Comment:
maybe `describeCluster` -> `describeClusterResult` is better?

Reply:
Done.